### ISSUE TITLE
[BUG] "Message absent from node" on ads

### ISSUE THREAD

**Issue by martudev** [WhiskeySockets/Baileys#1034]
**Describe the bug**
Lately I've been receiving many logs with the error message "Message absent from node" which, according to what I've seen, comes mainly from Facebook advertising. As I've seen, this error was fixed before but it keeps happening. I also see that on WhatsApp Web it says "waiting for messages..." and sometimes even when viewing it from my cell phone and answering from there it doesn't reach WhatsApp Web. My question is, would there be any way to fix it or do a fetch to get the message? Thanks!
Here its the message log
```
{
    "key": {
        "remoteJid": "<number>@s.whatsapp.net",
        "fromMe": false,
        "id": "3A930663A..."
    },
    "messageTimestamp": 17258854688,
    "pushName": "Karina",
    "broadcast": false,
    "messageStubType": 2,
    "messageStubParameters": [
        "Message absent from node"
    ]
}
```

**To Reproduce**
1. Send a message from fb ad and sometimes see that gets this error

**Expected behavior**
Some retry system or fetch message system

**Comment by wilsinho8** [comment]
https://github.com/WhiskeySockets/Baileys/pull/732

**Comment by martudev** [comment]
@wilsinho8 Thanks for the reply! I recently tried that change and it didn't work for me, if you look at the log I passed above, for some reason I don't know, my messageStubParameters only has one parameter, instead of 2, therefore when I do messageStubParameters[1] it returns undefined and throws an error when doing JSON.parse. Has something similar happened to you? Thanks!!

**Comment by wilsinho8** [comment]
> @wilsinho8 Thanks for the reply! I recently tried that change and it didn't work for me, if you look at the log I passed above, for some reason I don't know, my messageStubParameters only has one parameter, instead of 2, therefore when I do messageStubParameters[1] it returns undefined and throws an error when doing JSON.parse. Has something similar happened to you? Thanks!!

Detected message absent from node!!
- Message: {"key":{"remoteJid":"number@s.whatsapp.net","fromMe":false,"id":"3EB0633C0283EF0B8B1727"},"messageTimestamp":1726152174,"pushName":"Jobson","broadcast":false,"messageStubType":2,"messageStubParameters":["Message absent from node"]}
[Thu Sep 12 2024 14:43:00 GMT+0000 (Coordinated Universal Time)] Message ack not sent. Reason: SyntaxError: Unexpected token u in JSON at position 0

It happened with me too!
So far, no solution for this problem.
I recommend you join the Baileys discord to keep up with the news

**Comment by CSFelix** [comment]
I'm facing the same issue. `messageStubParameters` contains only one element

**Comment by PurpShell** [comment]
> I'm facing the same issue. `messageStubParameters` contains only one element

are you on the latest version? 

Are all of you guys looking at the message upsert array?
There should be more than one message

**Comment by CSFelix** [comment]
> > I'm facing the same issue. `messageStubParameters` contains only one element
> 
> are you on the latest version?
> 
> Are all of you guys looking at the message upsert array? There should be more than one message

Yep, I'm using `6.7.7` Baileys Version and it's into `messages.upsert` event listener

**Comment by wilsinho8** [comment]
> > I'm facing the same issue. `messageStubParameters` contains only one element
> 
> are you on the latest version?
> 
> Are all of you guys looking at the message upsert array? There should be more than one message

Yeah, i'm on 6.7.7 and error still persist.. 
I'm testing this pull request : https://github.com/WhiskeySockets/Baileys/pull/1043 .. to see if the problem be solved

By the way,
Purp take a look pls at 105
https://github.com/WhiskeySockets/Baileys/issues/105

**Comment by jfilipe-dev** [comment]
Same issue here, I'm using 6.7.9, `"messageStubParameters":["Message absent from node"]`

**Comment by piLovr** [comment]
Same Issue here, happened once! on a One-View Message, but the retry by replying to the Message worked. Sus.

**Comment by viniciussricci** [comment]
I am encountering an issue similar to the one fixed in PR #732 , where `"Message absent from node"` is returned. In that fix, the solution was to retrieve the original message (expected as the second element of `messageStubParameters`) and use `sendMessageAck` so that the message is delivered correctly.

In my case, I am receiving the following object:

```json
{
  "key": {
    "remoteJid": "xxxxxxxxxx@s.whatsapp.net",
    "fromMe": false,
    "id": "3EB0703C6FA391261415A54B8EF6C17476D37F90",
    "participant": undefined
  },
  "messageTimestamp": 1746122288,
  "pushName": "Leadsmart Setor de Testes",
  "broadcast": false,
  "messageStubType": 2,
  "messageStubParameters": [
    "Message absent from node"
  ]
}
```

Notice that `messageStubParameters` does not include the original message. I observed that this only happens when the message is sent from another application that uses Baileys. Could the absence of the second parameter be related to that scenario?

**Comment by rafaelfix** [comment]
@viniciussricci any luck yet?

I experience this issue when recieving messages from Meta's oficcial web.whastapp.com, not (only) from baileys.

**Comment by viniciussricci** [comment]
> [@viniciussricci](https://github.com/viniciussricci) any luck yet?
> 
> I experience this issue when recieving messages from Meta's oficcial web.whastapp.com, not (only) from baileys.

I'm not seeing these cases anymore. When I received them, the original message was generated shortly afterward. Which version of Baileys are you using?
